python/cudf_polars/cudf_polars/dsl/expr.py

```diff
@@ -1540,7 +1540,17 @@ def collect_agg(self, *, depth: int) -> AggInfo:
             raise NotImplementedError("Nan propagation in groupby for min/max")
         (child,) = self.children
         ((expr, _, _),) = child.collect_agg(depth=depth + 1).requests
-        if self.request is None:
+        if self.name == "first":
+            request = plc.aggregation.nth_element(
+                0, null_handling=plc.types.NullPolicy.INCLUDE
+            )
+        elif self.name == "last":
+            request = plc.aggregation.nth_element(
+                -1, null_handling=plc.types.NullPolicy.INCLUDE
+            )
+        else:
+            request = self.request
+        if request is None:
```

Comment:
```suggestion
        elif (request := self.request) is None:
```
Optional: Should avoid checking an always false condition for the first two `if/elif` above.

Reply:
Hmm, cute, but I don't like the lack of `else` fallthrough branch from a style point of view (this would be easier to handle if pylibcudf had typing).

I reordered so the assignment to request is unconditional and then overridden in the two exceptional cases instead.